### PR TITLE
chore: Faster tests

### PR DESCRIPTION
**Pull request by goergisn**
## Changes
- Removing long waiting times
- Improving flakey tests and readability
- Introducing `XCTestCase+Wait` with helper methods to wait until a certain condition is met to replace unreliable `wait(for:)`
- Introducing `XCTestCase+RootViewController` with helper methods to setup the window.rootViewController, present on the root and speed up all animations for faster tests
- Replicating local environment to ensure proper snapshot testing 
  - https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md

### CODE REVIEW

File: Tests/Components Tests/Gift Card/GiftCardComponentTests.swift [hunk 1]
```diff
@@ -489,7 +489,7 @@ class GiftCardComponentTests: XCTestCase {
 
         UIApplication.shared.keyWindow?.rootViewController = sut.viewController
 
-        wait(for: .milliseconds(300))
+        wait(for: .milliseconds(50))
```
Comment (descorp) [inline Tests/Components Tests/Gift Card/GiftCardComponentTests.swift]: Just a thought - should we parametrise this?

```swift
let aMoment = DispatchTimeInterval.milliseconds(50)
wait(for: aMoment)
```

**Review by erenbesel — DISMISSED** [review]
Nice that 50 ms is enough!

**Review by erenbesel — DISMISSED** [review]

File: Tests/Components Tests/Gift Card/GiftCardComponentTests.swift (hunk 1 above)
Comment (goergisn) [reply Tests/Components Tests/Gift Card/GiftCardComponentTests.swift]: Sounds good - could be an extension of DispatchTimeInterval `wait(for: .aMoment)`
Was also thinking about abstracting it into something like this as an extension on `XCTestCase`
```
func setupRootViewController(_ viewController: UIViewController) {
    UIApplication.shared.keyWindow?.rootViewController = viewController
    UIApplication.shared.keyWindow?.layer.speed = 10.0 // 10x Faster animations
    wait(for: .aMoment) // Giving the viewController a moment to get presented
}
```

File: UITests/Components/QRCode/__Snapshots__/QRCodeActionComponentUITests/testUIConfigurationForUPIQRCode-iPhone12.upi.png
Comment (descorp) [inline UITests/Components/QRCode/__Snapshots__/QRCodeActionComponentUITests/testUIConfigurationForUPIQRCode-iPhone12.upi.png:1]: This change seems dramatic 🤔

Comment (goergisn) [reply UITests/Components/QRCode/__Snapshots__/QRCodeActionComponentUITests/testUIConfigurationForUPIQRCode-iPhone12.upi.png:1]: Yess, fixing it

**Pull request closed without merging**